### PR TITLE
Exclude browser stability tests for exports from that specific browser

### LINKED ISSUE #29737: Stop running wpt-chrome-dev-stability for Chromium exports
The value of running wpt-chrome-dev-stability for Chromium exports is very low, and when it fails we (the Chromium team) almost always ignore. Let's not run it for PRs created by the wpt export bot.

### PR DESCRIPTION
**Pull request by DanielRyanSmith**
Fixes #29737

### CODE REVIEW

File: tools/ci/tc/decision.py [hunk 1]
```diff
@@ -74,6 +74,17 @@ def get_run_jobs(event):
...
+
+    # Exclude browser stability tests for exports from that specific browser.
+    try:
+        if event["sender"]["login"] == "chromium-wpt-export-bot":
+            all_jobs.discard("wpt-chrome-dev-stability")
+        if event["sender"]["login"] == "moz-wptsync-bot":
+            all_jobs.discard("wpt-firefox-nightly-stability")
+    except KeyError:
+        # Just continue if the username cannot be pulled from the event.
+        logger.info("Unable to read username from event. Continuing.")
```
Comment (jgraham) [inline tools/ci/tc/decision.py:86]: I'm not even sure we need to log this, but if we are maybe make it debug level.

File: tools/ci/tc/decision.py [hunk 2]
```diff
@@ -74,6 +74,17 @@ def get_run_jobs(event):
     path_jobs = jobs.get_jobs(paths)
     all_jobs = path_jobs | get_extra_jobs(event)
     logger.info("Including jobs:\n * %s" % "\n * ".join(all_jobs))
+
+    # Exclude browser stability tests for exports from that specific browser.
+    try:
```
Comment (jgraham) [inline tools/ci/tc/decision.py:79]: Can we put this in a `filter_jobs` function, and call it before the `Including jobs` log line?

Comment (DanielRyanSmith) [reply tools/ci/tc/decision.py]: Great idea - implemented

File: tools/ci/tc/decision.py (hunk 1 above)
Comment (DanielRyanSmith) [reply tools/ci/tc/decision.py]: I changed it to a debug level log👌

File: tools/ci/tc/decision.py [hunk 3]
```diff
@@ -100,6 +102,19 @@ def get_extra_jobs(event):
     return jobs
 
 
+def filter_jobs(jobs: Set, event: Dict) -> None:
```
Comment (DanielRyanSmith) [inline tools/ci/tc/decision.py:105]: Flake8 is upset about this line not using typing.Set and typing.Dict, but these are [deprecated as of Python 3.9](https://docs.python.org/3/library/typing.html#typing.Set). I'm just going to remove the type hints.

**Review by foolip — APPROVED** [review]
Looking good, thanks @DanielRyanSmith!

**Comment by foolip** [comment]
I've renamed the PR since this now does both Chrome and Firefox.

**Pull request merged**